lib/workers/repository/update/pr/changelog/source.ts

```diff
@@ -185,11 +188,13 @@ export abstract class ChangeLogSource {
   private findTagOfRelease(
     version: allVersioning.VersioningApi,
     packageName: string,
+    depName: string,
     depNewVersion: string,
     tags: string[],
   ): string | undefined {
-    const regex = regEx(`(?:${packageName}|release)[@-]`, undefined, false);
-    const exactReleaseRegex = regEx(`${packageName}[@\\-_]v?${depNewVersion}`);
+    const releaseRegexPrefix = `^(?:${packageName}|${depName}|release)[@_-]v?`;
```

Comment:
Unlike [in release-notes](https://github.com/renovatebot/renovate/pull/30395/files#r1694023245), no `/` here since the regex is later used to strip the release name in order to test if only the new version remains. We already know the test will fail if there's a prefix, so this doesn't change that behavior. (If it's not desired behavior, we can properly address it in a separate pull request.)